homeassistant/auth/__init__.py

```diff
@@ -479,6 +492,34 @@ async def async_remove_refresh_token(
         for revoke_callback in callbacks:
             revoke_callback()
 
+    async def async_track_next_refresh_token_expiration(self) -> None:
+        """Initialise all token expiration scheduled tasks."""
+        next_expiration = dt_util.utcnow() + REFRESH_TOKEN_EXPIRATION
+        for token in await self._store.async_get_refresh_tokens():
+            if (
+                expire_at := token.expire_at
+            ) is not None and expire_at < next_expiration:
+                next_expiration = expire_at
+
+        async def _remove_expired_refresh_tokens(_: datetime | None = None) -> None:
+            expired_tokens: list[models.RefreshToken] = []
+            now = dt_util.utcnow()
+            for token in await self._store.async_get_refresh_tokens():
+                if (expire_at := token.expire_at) is not None and expire_at <= now:
+                    expired_tokens.append(token)
+            for expired_token in expired_tokens:
+                await self.async_remove_refresh_token(expired_token)
```

Comment:
```suggestion
            now = dt_util.utcnow()
            for token in await copy(self._store.async_get_refresh_tokens()):
                if (expire_at := token.expire_at) is not None and expire_at <= now:
                    await self.async_remove_refresh_token(expired_token)
```

using the `copy` module could be an alternativ here, but i'm not sure about it's usage at all

Reply:
Looks like your suggestion will await copy ?

Reply:
You could use a view `[:]` as well to avoid the copy

Reply:
have change it to
```python
            now = dt_util.utcnow()
            for token in list(await self._store.async_get_refresh_tokens()):
                if (expire_at := token.expire_at) is not None and expire_at <= now:
                    await self.async_remove_refresh_token(token)
            await self.async_track_next_refresh_token_expiration()
```

Reply:
I put a suggestion below